### ISSUE TITLE
[Bug] - GitVersion.yml - ignore part ignored

### ISSUE THREAD

**Issue by mirecad** [GitTools/GitVersion#2412]
Gitversion does not take _ignore_ part of GitVersion.yml into account.

## Expected Behavior
We should be able to see _ignore_ settings of GitVersion.yml by running command `gitversion /showconfig`

## Actual Behavior
`gitversion /showconfig` shows default values for _ignore_ section in GitVersion 5.3.7. (I tested on 5.0.1 and everything was OK)

```
ignore:
  sha: []
```

## Steps to Reproduce
```
git init

git commit -m "initial" --allow-empty

cat GitVersion.yml
mode: Mainline
branches: {}
ignore:
  sha: [794454f56b4e7a1867b15bc3834d7169ed01b752]
merge-message-formats: {}

gitversion /showconfig
```
contains:
```
ignore:
  sha: []
```
but should contain:
```
ignore:
  sha:
  - 794454f56b4e7a1867b15bc3834d7169ed01b752
```

## Your Environment
- Version Used: 5.3.7+Branch.master.Sha.b22e8726683a0997484b53c5efd54394e93b5306
- Operating System and version : Windows 10.0.18362.0
- GitVersion installed by: choco install GitVersion.Portable

**Comment by asbjornu** [comment]
I don't know, but this **may** be a bug in `/showconfig` and not with `ignore` itself, as we have tests for this:

https://github.com/GitTools/GitVersion/blob/d326a8822531d1a0990d73dd68f48d9d026c0c63/src/GitVersionCore.Tests/Configuration/IgnoreConfigTests.cs#L14-L48

Are you experiencing that GitVersion is actually not ignoring the specified commit when calculating the version number?

**Comment by mirecad** [comment]
I just checked. Ignore specific sha is not working for me even on older versions. Older versions show at least correct `/showconfig`. 

You can see bellow, that I put sha of my last commit to ignore section, but version numbers are computed based on this commit.

FYI: I'm using gitversion for the first time. 

```
​D:\working\gitversiontestPS>gitversion -version
5.3.7+Branch.master.Sha.b22e8726683a0997484b53c5efd54394e93b5306

​D:\working\gitversiontestPS>cat .\GitVersion.yml
mode: Mainline
branches: {}
ignore:
  sha: [35b6bec470538f6ecc677babd470da17afd73aae]
merge-message-formats: {}

​D:\working\gitversiontestPS>gitversion
{
  "Major":0,
  "Minor":1,
  "Patch":3,
  "PreReleaseTag":3,
  "PreReleaseTagWithDash":"",
  "PreReleaseLabel":"",
  "PreReleaseNumber":"",
  "WeightedPreReleaseNumber":60000,
  "BuildMetaData":"",
  "BuildMetaDataPadded":"",
  "FullBuildMetaData":"Branch.master.Sha.35b6bec470538f6ecc677babd470da17afd73aae",
  "MajorMinorPatch":"0.1.3",
  "SemVer":"0.1.3",
  "LegacySemVer":"0.1.3",
  "LegacySemVerPadded":"0.1.3",
  "AssemblySemVer":"0.1.3.0",
  "AssemblySemFileVer":"0.1.3.0",
  "FullSemVer":"0.1.3",
  "InformationalVersion":"0.1.3+Branch.master.Sha.35b6bec470538f6ecc677babd470da17afd73aae",
  "BranchName":"master",
  "EscapedBranchName":"master",
  "Sha":"35b6bec470538f6ecc677babd470da17afd73aae",
  "ShortSha":"35b6bec",
  "NuGetVersionV2":"0.1.3",
  "NuGetVersion":"0.1.3",
  "NuGetPreReleaseTagV2":"",
  "NuGetPreReleaseTag":"",
  "VersionSourceSha":"ebc37dd3908db1e5285ad4b228fdc5fc19086a5f",
  "CommitsSinceVersionSource":3,
  "CommitsSinceVersionSourcePadded":"0003",
  "CommitDate":"2020-09-28"
}

​D:\working\gitversiontestPS>git log
commit 35b6bec470538f6ecc677babd470da17afd73aae (HEAD -> master)
Author: Miroslav Adamec <miroslav.adamec@invia.cz>
Date:   Tue Sep 29 00:32:16 2020 +0200

    commit 4

commit ae496287478f79bb7eb9cc8e9017666a3b38010a
Author: Miroslav Adamec <miroslav.adamec@invia.cz>
Date:   Tue Sep 29 00:31:05 2020 +0200

    commit 3

commit a818015a90cd3233590ce31f1cefd3635b11db43
Author: Miroslav Adamec <miroslav.adamec@invia.cz>
Date:   Tue Sep 29 00:30:54 2020 +0200

    commit 2

commit ebc37dd3908db1e5285ad4b228fdc5fc19086a5f
Author: Miroslav Adamec <miroslav.adamec@invia.cz>
Date:   Tue Sep 29 00:30:38 2020 +0200

    commit 1
```

**Comment by senritsu** [comment]
Can confirm this happens with 5.3.7 dotnet global tool. I have specified both a date, and individual commit hashes. A certain merge still shows up as version sources when running with `/output buildserver`, despite being both older than the specified date, and excluded by hash specifically.

**Comment by sdybvik** [comment]
This bug was introduced in 5.3.6. 

With this in GitVersion.yml: 
```
ignore:
  commits-before: 2020-09-04T00:00:00
```

The result when running `gitversion /showConfig` is this in these versions: 

5.3.6: 
```
ignore:
  sha: []
```
In 5.3.5: 
```
ignore:
  commits-before:
    date-time: 2020-09-04T00:00:00.0000000
    utc-date-time: 2020-09-03T22:00:00.0000000Z
    local-date-time: 2020-09-04T00:00:00.0000000+02:00
    date: 2020-09-04T00:00:00.0000000
    day: 4
    day-of-week: Friday
    day-of-year: 248
    month: 9
    offset: 02:00:00
    ticks: 637347744000000000
    utc-ticks: 637347672000000000
    year: 2020
  sha: []
```

**Comment by asbjornu** [comment]
Thanks for making that discovery, @sdybvik! We'll have to dig through the commit log find the problematic commit then.

**Comment by arturcic** [comment]
@mirecad does this issue fixes also #2354

**Comment by mirecad** [comment]
Yes. Fixes also #2354.
